src/array/ops/sum.rs

```diff
@@ -0,0 +1,33 @@
+use std::sync::Arc;
+
+use arrow2;
+
+use crate::{array::DataArray, datatypes::*, error::DaftResult};
+
+use super::DaftNumericAgg;
+
+macro_rules! impl_daft_numeric_agg {
+    ($arrayT:ident) => {
+        impl DaftNumericAgg for &$arrayT {
```

Comment:
You use the DaftDataType instead of DataArray type for this to be clearer.

i.e `impl DaftNumericAgg for &DataArray<DataType>`